keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -291,9 +285,26 @@ def make_generate_function(self):
         if self.generate_function is not None:
```

Comment:
Couldn't we check whether the method exists at the callsite? Do we need this factory style method?

Reply:
We could, but this would be less in line with how `keras.Model` works. See [these lines of keras.Model](https://github.com/keras-team/keras/blob/cbbe8eb53001ccffe9f02d08ecdaf22997879d19/keras/engine/training.py#L1265-L1291). The end goal is to pull this entire function into a base class that feels similar to `keras.Model` in how it handles this stuff.

Reply:
Leaving this for now, but happy to keep discussing.